nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/nf-connection-details.js

```diff
@@ -496,10 +496,10 @@
                         var availableRelationships = connection.availableRelationships;
                         var selectedRelationships = connection.selectedRelationships;
 
-                        // show the available relationship if applicable
+                        // show the selected relationships if applicable
                         if (nfCommon.isDefinedAndNotNull(availableRelationships) || nfCommon.isDefinedAndNotNull(selectedRelationships)) {
-                            // populate the available connections
-                            $.each(availableRelationships, function (i, name) {
+                            // populate the selected connections
+                            $.each(selectedRelationships, function (i, name) {
                                 createRelationshipOption(name);
                             });
```

Comment:
We need to account for dynamic Relationships that are selected within this Connection but have since been deleted from the source Processor. Previously we would render all Relationships here and in the block below we would identify any selected relationships that are no longer available and assign them the `undefined` class. Since now we're only rendering the selected relationships we'll probably want to handle that here and remove the block below.

Note: There is admittedly a bug in the existing implementation because `createRelationshipOption` is not actually returning the new option. It will need to be updated like:

```
    var createRelationshipOption = function (name) {
        return $('<div class="available-relationship-container"></div>').append(
            $('<div class="relationship-name"></div>').text(name)).appendTo('#read-only-relationship-names');
    };
```

You should see the Relationship that was removed is rendered in a lighter color and in italic.

Reply:
Thanks Matt for catching this. I just pushed a new commit with the suggested changes.

![Screenshot 2024-04-01 at 09 50 11](https://github.com/apache/nifi/assets/11541012/6486eaec-68f9-4dcd-94f8-951d8d3b56de)
![Screenshot 2024-04-01 at 09 50 20](https://github.com/apache/nifi/assets/11541012/40de7704-7aea-4b43-95e9-338d1fcca485)